### DISCUSSION TITLE
Is it possible to use the system Caddy for an app's proxy? If so, how?

### DISCUSSION THREAD

**Discussion by smammy** [dokku/dokku#5535] [Q&A] [answered]
I have a mix of Dokku-managed and manually-managed apps on a system. The manually-managed apps are proxied through a web server installed directly on the system from distribution packaging. When that web server was nginx, the Dokku `nginx` plugin would add an additional vhost to my system nginx config for each Dokku-managed app.

I'm switching this system from nginx to Caddy. First I installed Caddy (non-containerized) and set up vhosts for my manually-managed apps. Then I tried switching a Dokku-managed app to using the `caddy` plugin. However, the `caddy` plugin tries to run its own containerized Caddy, which tries to bind to the same ports (80 and 443) as the system Caddy.

I would have expected the `caddy` plugin to talk to the system Caddy, adding a vhost for each app, just like the `nginx` plugin does.

How can I convince Dokku to use the system Caddy for an app's proxy? Or is there another approach that would let manually-managed apps and Dokku-managed apps share the same proxy? What am I missing?

Thanks,
-sam

**Comment by josegonzalez** [comment, marked as answer]
Our caddy integration only works with [caddy-docker-proxy](https://github.com/lucaslorentz/caddy-docker-proxy). We have no plans at this time to generate a `Caddyfile` for every app in the same way we do `nginx.conf` files. This is by design, as it's incredibly difficult to deal with containers failing, coming up with different IPs, etc., and that's been a source of numerous tickets in Dokku.

You can certainly start `caddy-docker-proxy` via a custom compose file listening on different ports, then proxy to it from your system caddy. You could also use something like the [service-proxy](https://github.com/dokku/service-proxy) to proxy from a container to the host for each manually managed app (assuming those arent in Docker for whatever reason) and then use just the Caddy from the container.

For apps that are Dockerized but are "manually managed", you may wish to use our `git:from-image` command to deploy the apps as Dokku apps instead.
